### PR TITLE
Extend the ProbeHelper for use with CloudAuditLogsSource probe

### PR DESCRIPTION
**Pull request by tommyreddad**
Extended the Probe Helper to work with the CloudAuditLogsSource probe.

## Proposed Changes
- The Probe Helper can now receive events of type cloudauditlogssource-probe
- The Probe Helper can now handle events sourced from a CloudAuditLogsSource
- Added a test for the CloudAuditLogsSource probe

### CODE REVIEW

**Comment by Harwayne** [comment]
/hold

Please fill in the PR description, removing any unneeded fields. Then unhold.

File: test/test_images/probe_helper/main.go [hunk 1]
```diff
@@ -59,6 +59,19 @@ The Probe Helper can handle multiple different types of probes.
 		 the object, and waits to tbe notified that the object has been deleted by a
 		 CloudStorageSource.
 
+4. CloudAuditLogsSource Probe
+
+	This probe has two steps in executed in sequence which are intended to test a
```
Comment (liu-cong) [inline test/test_images/probe_helper/main.go:64]:
```suggestion
	This probe has two steps executed in sequence which are intended to test a
```

File: test/test_images/probe_helper/probe_helper.go [hunk 2]
```diff
@@ -243,6 +247,35 @@ func (ph *ProbeHelper) forwardFromProbe(ctx context.Context, brokerClient cloude
...
 			}
+		case CloudAuditLogsSourceProbeEventType:
+			switch event.Subject() {
+			case CloudAuditLogsSourceProbeCreateSubject:
+				channelID = event.ID() + "-" + event.Subject()
+				receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
+				if err != nil {
+					return logNACK(ctx, "Probe forwarding failed, could not create receiver channel: %v", err)
+				}
+				defer receivedEvents.deleteReceiverChannel(channelID)
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:258]: The channel management logic can be moved before `switch event.Subject()`.

File: test/test_images/probe_helper/probe_helper.go [hunk 3]
```diff
@@ -143,7 +147,7 @@ func logReceive(ctx context.Context, ack bool, format string, args ...interface{
 	return cloudevents.NewReceipt(ack, format, args...)
 }
 
-func (ph *ProbeHelper) forwardFromProbe(ctx context.Context, brokerClient cloudevents.Client, pubsubClient cloudevents.Client, bucket *storage.BucketHandle, receivedEvents *receivedEventsMap) cloudEventsFunc {
+func (ph *ProbeHelper) forwardFromProbe(ctx context.Context, brokerClient cloudevents.Client, cePubsubClient cloudevents.Client, pubsubClient *pubsub.Client, bucket *storage.BucketHandle, receivedEvents *receivedEventsMap) cloudEventsFunc {
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:150]: nit: the clients can be moved to the ProbeHelper struct

File: test/test_images/probe_helper/probe_helper.go [hunk 4]
```diff
@@ -301,6 +334,25 @@ func (ph *ProbeHelper) receiveEvent(ctx context.Context, receivedEvents *receive
 				return logACK(ctx, "Failed to extract event ID from object name: %v", err)
 			}
 			channelID = channelID + "-" + CloudStorageSourceProbeDeleteSubject
+		case schemasv1.CloudAuditLogsLogWrittenEventType:
+			// The logged event type is held in the methodname extension.
+			if _, ok := event.Extensions()["methodname"]; !ok {
+				return logACK(ctx, "CloudEvent does not have extension methodname: %v", event)
+			}
+			// For creation and deletion of pubsub topics, the topic ID can be extracted
+			// from the event subject.
```
Comment (liu-cong) [inline test/test_images/probe_helper/probe_helper.go:343]: Please document an example event here

File: test/test_images/probe_helper/probe_helper.go (hunk 2 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go]: Agreed, fixed. This cleans up the code a bit. It's important to remember that, e.g., the (not yet merged) CloudSchedulerSource probe won't be channel-based and so we'll have to remember not to leak a channel for that probe.

File: test/test_images/probe_helper/probe_helper.go (hunk 3 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go]: I moved lots of things to the ProbeHelper struct and allow them to be set manually if desired.

File: test/test_images/probe_helper/probe_helper.go (hunk 4 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe_helper.go]: Added examples for each probe.

**Review by liu-cong — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**